File: php/src/net_tcp6_sender.php
<?php

set_time_limit(0);

const RECEIVER_ADDRESS = "::1";
const RECEIVER_PORT = 10000;

echo "TCP IPv6 stream socket sender".PHP_EOL;

$socket = @socket_create(AF_INET6, SOCK_STREAM, SOL_TCP);
if ($socket === false)
{
  exit("- socket_create() failed: ".socket_strerror(socket_last_error()).PHP_EOL);
}
echo "- socket created".PHP_EOL;

$result = @socket_connect($socket, RECEIVER_ADDRESS, RECEIVER_PORT);
if ($result === false)
{
  exit("- socket_connect() failed: ".socket_strerror(socket_last_error($socket)).PHP_EOL);
}
echo "- connected to [".RECEIVER_ADDRESS."]:".RECEIVER_PORT.PHP_EOL;

$peer_address = "";
$peer_port = 0;

$peer = @socket_getpeername($socket, $peer_address, $peer_port);
if ($peer === false)
{
  exit("- socket_getpeername() failed: ".socket_strerror(socket_last_error($msgsock)).PHP_EOL);
}

$local_address = "";
$local_port = 0;

$local = @socket_getsockname($socket, $local_address, $local_port);
if ($local === false)
{
  exit("- socket_getsockname() failed: ".socket_strerror(socket_last_error($msgsock)).PHP_EOL);
}
echo "- for [".$peer_address."]:".$peer_port." bound on [".$local_address."]:".$local_port.PHP_EOL;

if (isset($argv[1]))
{
  $msg = $argv[1]; // message is the first parameter, for example "stop" to stop the receiver
}
else
{
  $msg = "This is PHP message sent at ".date("j.n.Y")." in ".date("G:i:s");
}
echo "|".$msg."|".PHP_EOL;

$msg_length = strlen($msg); 
$sent_length = @socket_send($socket, $msg, $msg_length, 0);
if ($sent_length === false)
{
  exit("- socket_send() failed: ".socket_strerror(socket_last_error($socket)).PHP_EOL);
}
echo "- message ".$sent_length."B of ".$msg_length."B sent".PHP_EOL;

socket_close($socket);
echo "- socket closed".PHP_EOL;

?>

File: python/src/net_tcp6_sender.py
import socket
import sys
from datetime import datetime

RECEIVER_ADDRESS = "::1";
RECEIVER_PORT = 10000;

print("TCP IPv6 stream socket sender")

sendingSocket = socket.socket(family=socket.AF_INET6, type=socket.SOCK_STREAM)
print("- socket created")

receiver_address = (RECEIVER_ADDRESS, RECEIVER_PORT)

sendingSocket.connect(receiver_address)
print("- connected to [" + RECEIVER_ADDRESS + "]:" + str(RECEIVER_PORT))

(local_address, local_port, local_object3, local_object4) = sendingSocket.getsockname()
print("- for [" + RECEIVER_ADDRESS + "]:" + str(RECEIVER_PORT) + " bound on [" + local_address + "]:" + str(local_port))

if len(sys.argv) > 1:
    msg = sys.argv[1] # message is the first parameter, for example "stop" to stop the receiver
else:
    now = datetime.now()
    msg = "This is Python message sent at " + now.strftime("%d.%m.%Y") + " in " + now.strftime("%H:%M:%S")
print("|" + msg + "|")

buffer = str.encode(msg)

msg_length = len(msg)
sendingSocket.send(buffer)
print("- message " + str(msg_length) + "B sent")

sendingSocket.close();
print("- socket closed")
